lib/inherited_resources/polymorphic_helpers.rb

```diff
@@ -120,8 +120,11 @@ def parent_class
       #
       def parent
         if parent_type
-          p = instance_variable_get("@#{parent_type}")
-          p || instance_variable_set("@#{parent_type}", association_chain[-1])
+          if instance_variable_defined?("@#{parent_type}")
+            instance_variable_get("@#{parent_type}")
+          else
+            instance_variable_set("@#{parent_type}", association_chain[-1])
+          end
```

Comment:
Are these really equivalent? Previously, if the instance varaible was `nil`, we were setting it, now we're just returning `nil`, right?

Maybe something like the following would be safer to preserve the behavior?

```ruby
p = instance_variable_defined?("@#{parent_type}") && instance_variable_get("@#{parent_type}")
p || instance_variable_set("@#{parent_type}", association_chain[-1])
```

Reply:
Exactly! It might not have any real effect since the test suite didn't catch it but as you said, safe is best! :+1:

Reply:
Yes I agree! I made the change so this is ready.